src/callbackiterresult.ts

```diff
@@ -1,26 +1,33 @@
-import IterResult from './iterresult'
-import {
-  contains
-} from './helpers'
+import IterResult, { IterArgs } from './iterresult'
+import { contains } from './helpers'
 
 /**
  * IterResult subclass that calls a callback function on each add,
  * and stops iterating when the callback returns false.
  */
 export default class CallbackIterResult extends IterResult {
-  constructor (method, args, iterator) {
+  private iterator: (d: Date, len: number) => boolean
+
+  constructor (
+    method: string,
+    args: Partial<IterArgs>,
+    iterator: (d: Date, len: number) => boolean
+  ) {
     const allowedMethods = ['all', 'between']
     if (!contains(allowedMethods, method)) {
-      throw new Error('Invalid method "' + method +
-        '". Only all and between works with iterator.')
+      throw new Error(
```

Comment:
💛 optional: template string here

Reply:
I scrapped that bit altogether. Thanks, Typescript!